Upgrade to Docker 20.10.05

Hi 👋

Due to the fact that the Drone Docker plugin still uses Docker v19.03.8:

```#!sh
latest: Pulling from plugins/docker
3s
2	Digest: sha256:1088e4e26b87ac4c24948ec2d2a738cea1f9e4e9a7b81eef3b77eae41d57e978
3s
3	Status: Image is up to date for plugins/docker:latest
3s
4	+ /usr/local/bin/dockerd --data-root /var/lib/docker --host=unix:///var/run/docker.sock
4s
5	Registry credentials or Docker config not provided. Guest mode enabled.
6s
6	+ /usr/local/bin/docker version
6s
7	Client: Docker Engine - Community
6s
8	 Version:           19.03.8
...
```

One will run into the following error when using `make` and a `Makefile` in conjunctionn with an Alpine image 3.14+:

```#!sh
Step 9/21 : RUN make TAG=$TAG BUILD=$BUILD build
83s
169	 ---> Running in 5371b7f93f10
83s
170	make: rm: Operation not permitted
84s
171	make: *** [Makefile:33: clean] Error 127
```

This was filed as a [bug report](https://wiki.alpinelinux.org/wiki/Release_Notes_for_Alpine_3.14.0#faccessat2) upstream with Alpine which linked to [this set of fixes and work-arounds](https://wiki.alpinelinux.org/wiki/Release_Notes_for_Alpine_3.14.0#faccessat2).

The simplest fix is to upgrade the  Docker version used here.

Can we do that please?

--- a/docker/docker/Dockerfile.linux.amd64
+++ b/docker/docker/Dockerfile.linux.amd64
@@ -1,4 +1,4 @@
-FROM docker:20.10.8-dind
+FROM docker:20.10.9-dind
 
 ENV DOCKER_HOST=unix:///var/run/docker.sock
 

--- a/docker/docker/Dockerfile.linux.amd64
+++ b/docker/docker/Dockerfile.linux.amd64
@@ -1,4 +1,4 @@
-FROM docker:19.03.8-dind
+FROM docker:20.10.8-dind
 
 ENV DOCKER_HOST=unix:///var/run/docker.sock
 

--- a/docker/docker/Dockerfile.linux.arm64
+++ b/docker/docker/Dockerfile.linux.arm64
@@ -1,4 +1,4 @@
-FROM arm64v8/docker:20.10.8-dind
+FROM arm64v8/docker:20.10.9-dind
 
 ENV DOCKER_HOST=unix:///var/run/docker.sock
 

--- a/docker/docker/Dockerfile.linux.arm64
+++ b/docker/docker/Dockerfile.linux.arm64
@@ -1,4 +1,4 @@
-FROM arm64v8/docker:19.03.8-dind
+FROM arm64v8/docker:20.10.8-dind
 
 ENV DOCKER_HOST=unix:///var/run/docker.sock
 

